jbehave-support-core/src/main/java/org/jbehavesupport/core/report/XmlReporterFactory.java

```diff
@@ -157,9 +160,47 @@ private void copyResourceToReportDirectory(Resource resource) {
     private void copyResourceToReportDirectory(Resource resource, String targetFilePath) {
         File targetFile = new File(targetFilePath);
         try {
-            FileUtils.copyInputStreamToFile(resource.getInputStream(), targetFile);
+            String resourcePath = String.valueOf(resource.getURI());
+            String jarPath = resourcePath.replace(("!/" + resource.getFilename()), "").replace("jar:file:", "");
+            File jarFile = new File(jarPath);
+
+            if (jarFile.isFile()) {
+                copyResourceFromJar(resource, targetFilePath, targetFile, jarFile);
+            } else {
+                copyResourceFromFile(resource, targetFile);
+            }
         } catch (IOException e) {
-            log.error("Cannot copy resource file to target", e);
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void copyResourceFromFile(Resource resource, File targetFile) throws IOException {
+        if (resource.isFile() && resource.getFile().isDirectory()) {
+            FileUtils.copyDirectory(resource.getFile(), targetFile);
+        } else {
+            FileUtils.copyInputStreamToFile(resource.getInputStream(), targetFile);
+        }
+    }
+
+    private void copyResourceFromJar(Resource resource, String targetFilePath, File targetFile, File jarFile) throws IOException {
+        try (JarFile jar = new JarFile(jarFile)) {
+            JarEntry entry = jar.getJarEntry(resource.getFilename());
+            if (entry.isDirectory()) {
+                jar.stream()
+                    .filter(file -> file.getName().startsWith(entry.getName()) && file.getName().endsWith(".xslt"))
+                    .forEach(file -> {
+                        String fileName = file.getName().replace(entry.getName(), "");
+                        File newFile = new File(targetFilePath + File.separator + fileName);
+                        try {
+                            FileUtils.copyInputStreamToFile(jar.getInputStream(file), newFile);
```

Comment:
In this case stream brings more problems than improves readability. I suggest to refactor it to iterator approach.